lightning/src/ln/functional_tests.rs

```diff
@@ -2632,12 +2632,13 @@ fn test_htlc_on_chain_success() {
 	let mut first_claimed = false;
 	for event in events {
 		match event {
-			Event::PaymentSent { payment_preimage } => {
-				if payment_preimage == our_payment_preimage {
+			Event::PaymentSent { payment_preimage, payment_hash } => {
+				if payment_preimage == our_payment_preimage && payment_hash == Some(_payment_hash) {
```

Comment:
nit: Could you rename the `_payment_hash` variable to `payment_hash_1`? (since a preceding underscore `_` usually means "unused") Same for `_payment_hash_2` below